### DISCUSSION TITLE
Batch script for alternating between selfhosted and public servers

### DISCUSSION THREAD

**Discussion by gio-giorgio** [rustdesk/rustdesk#6164] [Q&A]
Hello,

I am trying to create a batch script that allows me to alternate between my selfhosted server, for my devices connected through a VPN, and public servers that external clients will be using.

I got a tip from a Discord user, and came up with this:

```
ECHO OFF
CLS
:MENU
ECHO.
ECHO ...............................................
ECHO PRESS 1, or 2 to select your task, or 3 to EXIT.
ECHO ...............................................
ECHO.
ECHO 1 - Selfhosted
ECHO 2 - Public
ECHO 3 - EXIT
ECHO.
SET /P M=Type 1, 2, or 3 then press ENTER:
IF %M%==1 GOTO Selfhosted
IF %M%==2 GOTO Public
IF %M%==3 GOTO EOF
:Selfhosted
set "rustdesk_cfg=QfREDACTEDJye"
cd "C:\Program Files\RustDesk\"
RustDesk.exe --config %rustdesk_cfg%
exit
:Public
set "rustdesk_cfg="
cd "C:\Program Files\RustDesk\"
RustDesk.exe
exit
```

But it won't work. The selfhosted version does not even open RustDesk itself, but the next time I launch it, it contains the selfhosted network information. But that information remains until I remove it manually, from the GUI.

Is there any way I can improve this?

Thank you.

**Comment by DMSnz** [comment]
I would expect for your first issue, that RustDesk is already running (in the tray) when you run the batch file.
I would then expect you need to stop the RustDesk process/services and then run the batch script, or stop / restart RustDesk after running the --config part
Second part (without having used rustdesk --config command), I would expect you can not use NOTHING as a config parameter, so its likely skipping that and leaving you with self hosted settings
Not sure if that is correct but all I could think of..
You might need 1 config file to reference for self hosted and another for public, so each call in the script picks up the correct config as such.
